src/Microsoft.ApiDesignGuidelines.Analyzers/UnitTests/EnumStorageShouldBeInt32Tests.cs

```diff
@@ -12,12 +12,214 @@ public class EnumStorageShouldBeInt32Tests : DiagnosticAnalyzerTestBase
     {
         protected override DiagnosticAnalyzer GetBasicDiagnosticAnalyzer()
         {
-            return new BasicEnumStorageShouldBeInt32Analyzer();
+            return new EnumStorageShouldBeInt32Analyzer();
         }
 
         protected override DiagnosticAnalyzer GetCSharpDiagnosticAnalyzer()
         {
-            return new CSharpEnumStorageShouldBeInt32Analyzer();
+            return new EnumStorageShouldBeInt32Analyzer();
         }
+
+        #region CSharpUnitTests
+
+        [Fact]
+        public void CSharp_CA1028_NoDiagnostic()
+        {
+            VerifyCSharp(@"
+using System;
+namespace Test
+{
+    public enum TestEnum1 //no violation - because underlying type is Int32
+    {
+        Value1 = 1,
+        Value2 = 2
+    }
+    public static class OuterClass
+    {
+        [Flags]
+        public enum TestEnum2 : long //no violation - because underlying type is Int64 and has Flag attributes
+        {
+            Value1 = 1,
+            Value2 = 2,
+            Value3 = Value1 | Value2
+        }
+        private enum TestEnum3 : byte //no violation - because accessibility is private 
+        {
+            Value1 = 1,
+            Value2 = 2
+        }
+        internal class innerClass
+        {
+            public enum TestEnum4 : long //no violation - because resultant accessibility is private 
+            {
+                Value1 = 1,
+                Value2 = 2
+            }
+        }
+        public static void Main()
+        {
+        }
+    }
+}
+ ");
+        }
+
+        [Fact]
+        public void CSharp_CA1028_Diagnostic1()
```

Comment:
Provide meaningful test method names. "Diagnostic1" doesn't tell you what the test is about when you're looking at a failure in Test Explorer or on the console output. For example: `CSharp_CA1028_DiagnosticForInt64WithNoFlags`.